cmd/jujud/agent/machine/manifolds.go

```diff
@@ -1002,7 +1002,6 @@ func CAASManifolds(config ManifoldsConfig) dependency.Manifolds {
 		upgraderName: caasupgrader.Manifold(caasupgrader.ManifoldConfig{
 			AgentName:            agentName,
 			APICallerName:        apiCallerName,
-			UpgradeStepsGateName: upgradeStepsGateName,
```

Comment:
I don't think this is right. A caas controller, like an iaas one, needs to run any upgrade steps prior to starting the upgrade worker and potentially handling another upgrade, right? ie we'd want to do any database changes for the current upgrade steps before a potential restart for the next upgrade

Reply:
That is not what this does though AFAICT.

It is not used or checked by the worker; it is only an input, which as I understand would simply restart this worker if it changes.

But this worker doesn't need to restart when the upgrade-steps worker opens the gate. It is in a loop looking for version changes all the time.

The problem is that it *does* operate and (at least for IAAS) can pick up new versions, taking down agents while upgrade steps are running.

My plan is to take this out, take the upgrade check flag out of `isFullyUpgraded`, which we can then wrap this with without deadlocking it.